internal/label/label.go

```diff
@@ -13,7 +13,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package resolve
+package label
```

Comment:
This make the typename label.Lablel
The traditional approach here is to rename the type to a single letter (label.L)

Also label.ParseLabel should just be label.Parse

Reply:
@alandonovan Can you settle a readability question? Does a single letter type name (L) make sense when the type would otherwise have the same name as the package (label.Label)? This seems overly brief to me, but Ian points out this policy works well for testing.T and should have been applied to context.Context. WDYT?

Reply:
There are plenty of precedents for "single-type packages" whose name is the same as their sole type (html/template, math/rand, context, etc).  I think label.Label is fine, as is template.Template, though perhaps context.Context should have followed testing.T if only for the frequency with which users have to write this name.

Reply:
Decided to keep `label.Label`. That feels more readable to me.

Renamed `label.ParseLabel` to `Parse` and added `New` constructor.